pulser-core/pulser/sequence/_seq_drawer.py

```diff
@@ -707,11 +726,136 @@ def phase_str(phi: float) -> str:
+            phase values are ommited.
+        draw_phase_shifts: Whether phase shift and reference information
+            should be added to the plot, defaults to False.
+        draw_input: Draws the programmed pulses on the channels, defaults
+            to True.
+        draw_modulation: Draws the expected channel output, defaults to
+            False. If the channel does not have a defined 'mod_bandwidth', this
+            is skipped unless 'draw_input=False'.
+        draw_phase_curve: Draws the changes in phase in its own curve (ignored
+            if the phase doesn't change throughout the channel).
+    """
+    (fig_reg, fig, ch_axes, data) = _draw_channel_content(
+        sampled_seq,
+        register,
+        sampling_rate,
+        draw_phase_area,
+        draw_phase_shifts,
+        draw_input,
+        draw_modulation,
+        draw_phase_curve,
+    )
+
+    return (fig_reg, fig)
+
+
+def draw_sequence(
+    seq: pulser.sequence.Sequence,
+    sampling_rate: Optional[float] = None,
+    draw_phase_area: bool = False,
+    draw_interp_pts: bool = True,
+    draw_phase_shifts: bool = False,
+    draw_register: bool = False,
+    draw_input: bool = True,
+    draw_modulation: bool = False,
+    draw_phase_curve: bool = False,
+) -> tuple[Figure | None, Figure]:
+    """Draws the entire sequence.
+
+    Args:
+        seq: The input sequence of operations on a device.
+        sampling_rate: Sampling rate of the effective pulse used by
+            the solver. If present, plots the effective pulse alongside the
+            input pulse.
+        draw_phase_area: Whether phase and area values need to be shown
+            as text on the plot, defaults to False. If `draw_phase_curve=True`,
+            phase values are ommited.
+        draw_interp_pts: When the sequence has pulses with waveforms of
+            type InterpolatedWaveform, draws the points of interpolation on
+            top of the respective waveforms (defaults to True).
+        draw_phase_shifts: Whether phase shift and reference information
+            should be added to the plot, defaults to False.
+        draw_register: Whether to draw the register before the pulse
+            sequence, with a visual indication (square halo) around the qubits
+            masked by the SLM, defaults to False.
+        draw_input: Draws the programmed pulses on the channels, defaults
+            to True.
+        draw_modulation: Draws the expected channel output, defaults to
+            False. If the channel does not have a defined 'mod_bandwidth', this
+            is skipped unless 'draw_input=False'.
+        draw_phase_curve: Draws the changes in phase in its own curve (ignored
+            if the phase doesn't change throughout the channel).
+    """
+    # Sample the sequence and get the data to plot
+    shown_duration = seq.get_duration(include_fall_time=draw_modulation)
+    sampled_seq = sample(seq, extended_duration=shown_duration)
```

Comment:
By extending it here, you know you will get different results from the previous implementation, because the duration of `sampled_seq` is not the same as the `seq` you were providing before.
Normally all the commands related to extension is handled in `gather_data`, why did you feel you had to put it here as well ? 
I suggest you take out `extended_duration=shown_duration`.